### ISSUE TITLE
Incorrect Handling of Media Filenames

### ISSUE THREAD

**Issue by E0697** [ankidroid/Anki-Android#16462]
### Checked for duplicates?

- [X] This issue is not a duplicate

### Does it also happen in the desktop version?

- [X] This bug does not occur in the latest version of Anki

### What are the steps to reproduce this bug?

1. Create a long screenshot in Brave for Android (from sharing hub)
2. The resulting file has a filename template that follows this example: chrome_screenshot_May 22, 2024 4_54_13 PM GMT+03_00
3. Add above file to a card
4. Use the media check tool > Image file is deemed unused

### Expected behaviour

The file shouldn't show up in the media check as unused. The check result should be empty (no  unreferenced cards).

### Actual behaviour

The file is detected as unreferenced, and isn't included in exports.

### Debug info

```text
AnkiDroid Version = 2.18.0 (464cf290cc77c3f2c562cb03b408b78a13be261b)

Backend Version = 0.1.38-anki24.04.1 (24.04.1 ccd9ca1a8309b80bcb50ddc5d99c7ce63440bce9)

Android Version = 14 (SDK 34)

ProductFlavor = full

Manufacturer = realme

Model = 7 Pro

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 14; 7 Pro Build/UQ1A.240205.004; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/124.0.6367.179 Mobile Safari/537.36

ACRA UUID = e30b1244-b08d-4fdd-ad9a-a898dde99cee

FSRS Enabled = false

Crash Reports Enabled = true
```

### (Optional) Anything else you want to share?

I tried changing the filename to 'chrome' and it worked properly, tried removing spaces and some other characters but it didn't help.

Here's a sample deck (single card) and the media file seperately: 
[Deck+Media.zip](https://github.com/ankidroid/Anki-Android/files/15415950/Deck%2BMedia.zip)

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by E0697** [comment]
Might be related: https://github.com/ankidroid/Anki-Android/pull/15890#issue-2185923304

**Comment by david-allison** [comment]
Reproduced this in Anki Desktop

<img width="800" alt="Screenshot 2024-06-05 at 19 22 03" src="https://github.com/ankidroid/Anki-Android/assets/62114487/8326fa54-a4b5-4432-a79a-047fcdedcc2c">

[16462.colpkg.zip](https://github.com/user-attachments/files/15594858/16462.colpkg.zip)

Please report this as a bug in Anki Desktop

**Comment by mikehardy** [comment]
I think that means we can actually *close* here correct? As it will just come in automatically once we ingest whatever version of desktop gets the fix?

**Comment by david-allison** [comment]
Sure - ideally with a reference to the upstream issue

**Comment by E0697** [comment]
But did you try to recreate the note on the desktop client? Or was it the same one I uploaded?

**Comment by E0697** [comment]
I believe the problem is with the way AnkiDroid encodes the media filenames, due to a recent change that is. A simple space in the filename will reproduce the big. It's reproducible on the desktop apps due to the aforementioned improper encoding. Creating the note via a desktop app doesn't cause any issues, only importing the one created via AnkiDroid does.

`The following files are referenced by cards, but were not found in the media folder:
Missing: chrome_screenshot_May%2022%2C%202024%204_54_13%20PM%20GMT%2B03_00.png

The following files were found in the media folder, but do not appear to be used on any cards:
Unused: chrome_screenshot_May 22, 2024 4_54_13 PM GMT+03_00.png`

**Comment by E0697** [comment]
It's true that it's reproducible on the desktop apps if you import a note created in AnkiDroid. If you create the note on the desktop app however, it works properly. Importing a note created in the desktop app also works without any issues on AnkiDroid.
It seems that AnkiDroid is storing the filename "example img.png" as "example%20img.png". Replacing the %20 with a space seems to fix the problem. That's also how it shows up in the note created on desktop.

**Comment by acontal** [comment]
I am seeing the same issue with media file names that contain Japanese characters.

Steps to reproduce:
1. Download the image at https://ja.wikipedia.org/wiki/%E3%83%95%E3%82%A1%E3%82%A4%E3%83%AB:%E6%97%AD%E6%97%A5%E6%97%97.png
2. Add the image to an AnkiDroid card.
3. The card now has "<img src="%E6%97%AD%E6%97%A5%E6%97%97.png">" in it.
4. Save the card.
5. Tap on "Check media".

Expected: AnkiDroid does not find any unused media file.
Actual: AnkiDroid says: "Unused files: 旭日旗.png"

It seems that the code that stores the image in the local file system, and the code that checks for usage are not using the same URI-encoding strategy.

**Comment by acontal** [comment]
If I rewrite manually the media file name in the card to its original form (i.e., "旭日旗.png"), everything works well: the image gets displayed when rendering the card, and the media check feature does not complain about an unused media file. It looks like the feature that adds media files to fields should not do any URI-encoding, as it is incompatible with the way the media checker works, and it also makes the media name unreadable for humans.

**Comment by E0697** [comment]
The latest beta renames the image files (similar to cropped images naming scheme) when adding them. That pretty much solves it for newly added files. I think the files added with the previous encoding need to be renamed manually. I probably should close this now?

**Comment by E0697** [comment]
@david-allison Can you please read some of the newer comments. It seems the issue has to do with how AnkiDroid encodes the filenames.

**Comment by david-allison** [comment]
@E0697 the issue is still open

**Comment by E0697** [comment]
Indeed. Sorry for tagging. I just thought it wasn't currently in consideration because of the upstream-issue label.

**Comment by ravirpsc** [comment]
I've also had this issue for long. I first noticed this when I created more than 50 cards with images on my android tablet and synced my phone but all the images were missing on my phone. I did some trial & error and found out that those files with spaces or some special characters in their in their names would automatically get these characters replaced by a '%' and tagged unused because the media checker would search for these automated new names instead of the original one. Also the AnkiDroid media folder on my tablet had the original name files so they would show properly but still be tagged as unused somehow.

For example: 07) Winds, Cyclones & Ocean Currents.jpg (won't work ❌)
If I rename to: 07-Winds_Cyclones_Ocean_Currents.jpg (will work ✅)

**Comment by E0697** [comment]
Still searching for solutions!
